raw_bsd.go

```diff
@@ -111,13 +117,38 @@ func listenPacket(ifi *net.Interface, proto Protocol) (*packetConn, error) {
 
 // ReadFrom implements the net.PacketConn.ReadFrom method.
 func (p *packetConn) ReadFrom(b []byte) (int, net.Addr, error) {
+	var timeout time.Duration
+	var hasTimeout bool
+
+	p.timeoutMu.Lock()
+	if !p.rtimeout.IsZero() {
+		timeout = p.rtimeout.Sub(time.Now())
+		hasTimeout = true
+	}
+	p.timeoutMu.Unlock()
+
+	if hasTimeout && timeout < time.Microsecond {
+		return 0, nil, &timeoutError{}
+	}
+
+	tv := &unix.Timeval{
+		Sec:  int64(timeout / time.Second),
+		Usec: int64(timeout % time.Microsecond),
+	}
+
+	if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, uintptr(p.fd), syscall.BIOCSRTIMEOUT, uintptr(unsafe.Pointer(tv))); err != 0 {
```

Comment:
This seems like an unusual API.  Any reason you didn't stick with poll or something?

Reply:
`poll` is more complex than just setting a read timeout. `poll` is helpful if you have multiple file descriptors and/or multiple events you want to be notified about. There is also a `SetBpfTimeout` function in the standard library, but these BPF functions are going to be removed. In my point of view setting a timeout by using `BIOCSRTIMEOUT` is the simplest way. It's documented here:
https://www.freebsd.org/cgi/man.cgi?query=bpf&sektion=4